source/Octopus.Tentacle/Communications/HalibutInitializer.cs

```diff
@@ -133,6 +134,17 @@ IPEndPoint GetEndPointToListenOn()
 
         public void Stop()
         {
+            halibut.DisposeAsync().GetAwaiter().GetResult();
+        }
+
+        public async ValueTask DisposeAsync()
```

Comment:
DisposeAsync is not called when the container is disposed of, but this has been added for completeness.